### ISSUE TITLE
niffler dependent on failure

### ISSUE THREAD

**Issue by Ebedthan** [luizirber/niffler#45]
Hi,

After RUSTEC pointed out that [failure is deprecated](https://github.com/rust-lang-nursery/failure/pull/347), I've found doing `crate tree` inside my project that niffler is dependent on the failure crate. Is there a way to get rid of failure? 

Thanks in advance.

**Comment by luizirber** [comment]
Hi @Ebedthan! `failure` is an indirect dependency (`bgzip` uses it). If you want to avoid it, you can select what formats you want to support explicitly using features, as pointed in the [README](https://github.com/luizirber/niffler#selecting-compression-formats):
```
niffler = { version = "2.3.1", default-features = false, features = ["bz2", "lzma", "gz"] }
```

**Comment by Ebedthan** [comment]
Thanks for pointing this out. Closing this issue and contacting bgzip maintainer.

**Comment by Ebedthan** [comment]
Hi @luizirber ,
Following my PR on bgzip repo, we have updated bgzip to turn from failure to thiserror and released v0.2.0 on crate.io. So you need to update your cargo.toml to get rid of failure. 
Thank you everyone.

**Comment by luizirber** [comment]
Thanks! `2.3.2` released, enjoy =]
